### PR TITLE
fix(authentication)!: add server specific issuer to JWT token (DEV-555)

### PR DESCRIPTION
**Pull request by irinaschubert**
Resolves DEV-555

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Review by subotic — APPROVED** [review]

File: sipi/scripts/jwt.lua [hunk 1]
```diff
@@ -14,8 +14,22 @@ function get_knora_token()
...
+    if webapi_port == nil then
+        send_error(500, "KNORA_WEBAPI_KNORA_API_EXTERNAL_PORT not set")
+        return nil
+    end
+    
+    token_issuer = webapi_hostname .. ':' .. webapi_port
+    server.log("token_issuer:" .. token_issuer, server.loglevel.LOG_DEBUG)
+    if token["iss"] ~= token_issuer then
+        server.log(token_issuer, server.loglevel.LOG_DEBUG)
+        send_error(401, "Invalid token. The request came from a different server than the one issued the token.")
```
Comment (BalduinLandolt) [inline sipi/scripts/jwt.lua:32]:
```suggestion
        send_error(401, "Invalid token. The request came from a different server than the one that issued the token.")
```
... or maybe something along the lines of "The token was not issued by the same server that sent the request".

File: webapi/src/test/scala/org/knora/webapi/TestContainersAll.scala [hunk 2]
```diff
@@ -36,18 +36,21 @@ object TestContainersAll {
...
-  SipiContainer.withCommand("--config=/sipi/config/sipi.knora-docker-config.lua")
+
+  SipiContainer.withCommand("--config=/sipi/config/sipi.docker-config.lua")
 
   // TODO: Needs https://github.com/scalameta/metals/issues/3623 to be resolved
   SipiContainer.withClasspathResourceMapping(
-    // "/sipi/config/sipi.knora-docker-config.lua"
-    "/sipi.knora-docker-config.lua",
-    "/sipi/config/sipi.knora-docker-config.lua",
+    // "/sipi/config/sipi.docker-config.lua"
```
Comment (BalduinLandolt) [inline webapi/src/test/scala/org/knora/webapi/TestContainersAll.scala:51]: are those comments still valid and needed?

**Review by BalduinLandolt — APPROVED** [review]
looks good to me

**Review by mpro7 — APPROVED** [review]

File: webapi/src/test/scala/org/knora/webapi/TestContainersAll.scala (hunk 2 above)
Comment (irinaschubert) [reply webapi/src/test/scala/org/knora/webapi/TestContainersAll.scala:51]: yes, they belong to the TODO which is still open (we have to wait for a fix in the metals plugin)

**Pull request merged**